readthedocs/projects/tasks/builds.py

```diff
@@ -206,7 +192,16 @@ def execute(self):
     bind=True,
 )
 def sync_repository_task(self, version_id):
-    self.execute()
+    lock_id = f"{self.name}-lock-{self.data.project.slug}"
+    with memcache_lock(lock_id, 60, self.app.oid) as lock_acquired:
```

Comment:
I'd like to see kwargs for this, since it's really not obvious what the arguments here are doing.